src/utilities/geometry/Intersection.cpp

```diff
@@ -49,22 +50,54 @@
 #include <boost/geometry/strategies/cartesian/point_in_poly_crossings_multiply.hpp>
 #include <boost/geometry/algorithms/within.hpp>
 #include <boost/geometry/algorithms/simplify.hpp>
+#include <boost/geometry/policies/robustness/get_rescale_policy.hpp>
+#include <boost/geometry/policies/robustness/robust_point_type.hpp>
 #if defined(_MSC_VER)
 #  pragma warning(pop)
 #endif
 
-typedef double coordinate_type;
-typedef boost::geometry::model::d2::point_xy<double> BoostPoint;
-typedef boost::geometry::model::polygon<BoostPoint> BoostPolygon;
-typedef boost::geometry::model::ring<BoostPoint> BoostRing;
-typedef boost::geometry::model::multi_polygon<BoostPolygon> BoostMultiPolygon;
+using coordinate_type = double;
+using BoostPoint = boost::geometry::model::d2::point_xy<double>;
+using BoostPolygon = boost::geometry::model::polygon<BoostPoint>;
+using BoostRing = boost::geometry::model::ring<BoostPoint>;
+using BoostMultiPolygon = boost::geometry::model::multi_polygon<BoostPolygon>;
 
 #include <polypartition/polypartition.h>
 
 namespace openstudio {
 
 // Private implementation functions
 
+// Boost 1.73.0 has this bit in has_self_intersections.hpp:
+//
+// For backward compatibility
+// template <typename Geometry>
+// inline bool has_self_intersections(Geometry const& geometry, bool throw_on_self_intersection = true) {
+//   typedef typename geometry::point_type<Geometry>::type point_type;
+//   typedef typename geometry::rescale_policy_type<point_type>::type rescale_policy_type;
+
+//   typename strategy::intersection::services::default_strategy<typename cs_tag<Geometry>::type>::type strategy;
+
+//   rescale_policy_type robust_policy = geometry::get_rescale_policy<rescale_policy_type>(geometry, strategy);
+
+//   return has_self_intersections(geometry, strategy, robust_policy, throw_on_self_intersection);
+// }
+
+// For backward compatibility
+template <typename Geometry>
+inline bool has_self_intersections(Geometry const& geometry, bool throw_on_self_intersection = true) {
+  using point_type = typename boost::geometry::point_type<Geometry>::type;
+  using rescale_policy_type = typename boost::geometry::rescale_policy_type<point_type>::type;
+
+  typename boost::geometry::strategies::relate::services::default_strategy<Geometry, Geometry>::type strategy;
+
+  // typename boost::geometry::strategy::intersection::services::default_strategy<typename boost::geometry::cs_tag<Geometry>::type>::type strategy;
+
+  auto robust_policy = boost::geometry::get_rescale_policy<rescale_policy_type>(geometry, strategy);
+
+  return boost::geometry::detail::overlay::has_self_intersections(geometry, strategy, robust_policy, throw_on_self_intersection);
+}
+
```

Comment:
An actual change. `has_self_intersections` overload we're using was finally removed in 1.76. implement workaround